src/main/java/io/airlift/slice/Slice.java

```diff
@@ -1206,7 +1206,7 @@ public ByteBuffer toByteBuffer(int index, int length)
         checkIndexLength(index, length);
 
         if (base instanceof byte[]) {
-            return ByteBuffer.wrap((byte[]) base, (int) ((address - ARRAY_BYTE_BASE_OFFSET) + index), length);
+            return ByteBuffer.wrap((byte[]) base, (int) ((address - ARRAY_BYTE_BASE_OFFSET) + index), length).slice();
```

Comment:
I pretty sure you get an extra allocation here also.

Reply:
Yes, it does an extra allocation of the `ByteBuffer` object. But it should hopefully be optimized away, and conversion is unlikely to happen in a hot path.